File: Arcade/Intro/2 - Edge of the Ocean/4 - adjacentElementsProduct/adjacentElementsProduct.java
int adjacentElementsProduct(int[] inputArray) {

    int largestProduct = inputArray[0]*inputArray[1];;
    
    for(int index = 1; index < inputArray.length-1; index++)
    {
        int result = inputArray[index]*inputArray[index+1];
        
        boolean foundBiggerValue = result > largestProduct;
        if(foundBiggerValue) largestProduct = result;
    }
    
    return largestProduct;
}

File: Arcade/Intro/2 - Edge of the Ocean/4 - adjacentElementsProduct/adjacentElementsProduct.rb
def adjacentElementsProduct(inputArray)
    
    max = -99999
    i = 0
    
    loop do
        found_a_bigger_number = inputArray[i]*inputArray[i+1] > max
        
        if(found_a_bigger_number)
            max = inputArray[i]*inputArray[i+1]
        end
        i += 1
        break if i+1 == inputArray.length
    end
    
    return max
end
